### DISCUSSION TITLE
How to address a loaded sql file that is too large to process

### DISCUSSION THREAD

**Discussion by CordThomas** [pawelsalawa/sqlitestudio#4854] [Q&A] [answered]
### Details
Here you describe your problem details, or proposal for enhancement.

I foolishly loaded a 250MB sql file to recreate a database and the studio is unresponsive.   I have tried restarting without luck.  I have searched the settings3 configuration table but see no evidence of the 100,000 sql statements.

How do i correct my sqlite studio to remove that error of my judgement.

### Steps to reproduce
This applies only to bug reports. If this is a feature request, delete this section.

load a sqlite dump file over 250MB

### Operating system
This applies only to bug reports. If this is a feature request, delete this section.

Mac OSX Ventura 13.5

### SQLiteStudio version
This applies only to bug reports. If this is a feature request, delete this section.

I think it's 3.2.4 but can't see.

**Comment by tuffnatty** [comment]
If your problem is in automatic re-opening of the huge file, I would try to rename it or move to another location.

**Reply by CordThomas** [reply]
Hello, that did not work.   I renamed (and then tried moving) the file the editor window still shows the contents of the file.

**Comment by pawelsalawa** [comment, marked as answer]
```
delete from settings where [group] = 'General' and [key] = 'Session';
```

**Reply by pawelsalawa** [reply]
Also, to execute such huge SQL files in SQLiteStudio, please use right-click on the database and select "Execute SQL from file".

**Reply by CordThomas** [reply]
@pawelsalawa , I executed that statement and that solved the problem of the editor not opening but it also deleted all my editor tabs.  I had a dozen editor tabs for work on 6 different databases and now those are gone.  I guess I should have asked "where are the editor tabs represented in the settings table so that i can erase or remove one".   I can't for the life of me find "SQL editor 4" for example.  Are their contents recoverable somewhere?   This is not a disaster, but i have lost a few complex queries.

**Reply by pawelsalawa** [reply]
If you have ever executed these queries you will find them on the "History" tab of the SQL Editor window.

**Reply by CordThomas** [reply]
Thank you - could you tell me where the sql editor tabs are stored?

**Reply by pawelsalawa** [reply]
All tabs (not just editor tabs) are stored in a single cell (as serialized binary value) that you deleted earlier from settings table. They are stored alongside many other properties of your current application state - there are open windows/tabs, information about expanded/collapsed nodes of your databases tree, etc.

**Reply by CordThomas** [reply]
Please remind me where the SQLiteStudio db is so i can run that command again (delete from settings ...) as in my haste, I repeated my bungle.  Thank you.

**Reply by pawelsalawa** [reply]
Locations, depending on Operating System are documented at https://github.com/pawelsalawa/sqlitestudio/wiki/Configuration_directory_location

**Reply by CordThomas** [reply]
Thank you.
